packages/react/src/auto/hooks/useRelatedModelOptions.tsx

```diff
@@ -0,0 +1,246 @@
+  const modelManager = getModelManager(api, model.apiIdentifier, model.namespace);
+
+  const [{ data: selectedRecord, fetching: fetchingSelected, error: fetchSelectedRecordError }] = useFindOne(
+    modelManager as any,
+    currentRecordId ?? "",
+    {
+      pause: !currentRecordId, // BelongsTo needs a selected record to query in the related model
+      select: {
+        id: true,
+        [`${field}Id`]: true, // Retrieve the raw field value, regardless of if the ID exists or not
+        [field]: { _all: true }, // All of the fields on the related record iff the record exists
+      },
+    }
+  );
+
+  return {
+    selected: {
+      records: selectedRecord ? [selectedRecord] : undefined,
+      // records: selectedRecord && selectedRecord[field] ? [(selectedRecord[field] as any)._all] : undefined,
+      fetching: fetchingSelected,
+      error: fetchSelectedRecordError,
+    },
+  };
+};
+
+export const useLinkedParentModelRelatedModelRecords = (props: {
+  field: string; // Field API identifier
+}) => {
+  const { field } = props;
+  const { findBy } = useAutoFormMetadata();
+  const { metadata } = useFieldMetadata(field);
+  const { modelManager: relatedModelManager } = useRelatedModelManager({ field });
+
+  const fieldType = metadata.configuration.fieldType;
+  if (![FieldType.HasOne, FieldType.HasMany].includes(fieldType)) {
+    throw new Error(`Invalid field type "${fieldType}" used in useLinkedParentModelRelatedModelRecords`);
+  }
+
+  const inverseFieldApiId = (metadata.configuration as GadgetHasOneConfig | GadgetHasManyConfig).inverseField?.apiIdentifier;
+  if (!inverseFieldApiId) {
+    throw new Error(`The inverse field is not configured for the relationship field "${field}"`);
+  }
+
+  const currentRecordId = typeof findBy === "string" ? findBy : undefined;
+
+  const [{ data: selectedRecords, fetching: fetchingSelected, error: fetchSelectedRecordError }] = useFindMany(relatedModelManager as any, {
+    pause: !currentRecordId, // HasOne/HasMany need the current record to query the inverse field in the related model
+
+    first: selectedRecordsToLoadCount, // Many records can point to the current record in hasOne/hasMany
+    filter: { [inverseFieldApiId]: { equals: currentRecordId } }, // Filter by the inverse field belongsTo field value
+  });
+
+  return {
+    selected: {
+      records: selectedRecords,
+      fetching: fetchingSelected,
+      error: fetchSelectedRecordError,
+    },
+  };
+};
+
+export const useAllRelatedModelRecords = (props: {
+  field: string; // Field API identifier
+  optionLabel?: OptionLabel;
+}) => {
+  const { field, optionLabel } = props;
+  const optionLabelIsFieldName = typeof optionLabel === "string";
+
+  const {
+    modelManager: relatedModelManager,
+    model: { apiIdentifier, namespace, inverseFieldApiIdentifier },
+  } = useRelatedModelManager({ field });
+
+  const [loadedRecords, setLoadedRecords] = useState<any[]>([]);
```

Comment:
I think if someone clicks through enough pages this could get really big -- do you wanna keep them all loaded in memory or just the last loaded page? I don't feel strongly

Reply:
Polaris list only has a `onScrolledToBottom` callback, without a `scrollToTop` equivalent. Without something like this, if you scroll past a page containing something you want, you'd have to close->reopen the list to go back to a previous page. Doing this hurts more the deeper that the pages get, especially if the thing you want is hard to search for